Add split event processor for dataprepper (#6682)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/processors/split-event.md

---
layout: default
title: split-event
parent: Processors
grand_parent: Pipelines
nav_order: 56
---

# split-event

The `split-event` processor is used to split events based on a delimiter and generates multiple events from a user-specified field.

## Configuration

> The following table describes the configuration options for the `split-event`

Reviewer: ```suggestion
The following table describes the configuration options for the `split-event` processor.
```

| Option           | Type    | Description                                                                                   |
|------------------|---------|-----------------------------------------------------------------------------------------------|

> | `field`          | String  | The field in the event to be split.                                                           |

Reviewer: ```suggestion
| `field`          | String  | The event field to be split.                                                           |
```

| `delimiter_regex`| String  | The regular expression used as the delimiter for splitting the field.                         |
| `delimiter`      | String  | The delimiter used for splitting the field. If not specified, the default delimiter is used.  |

# Usage

To use the `split-event` processor, add the following to your `pipelines.yaml` file:

```
split-event-pipeline:
  source:
    http:
  processor:
    - split_event:
        field: query
        delimiter: ' '    
  sink:
    - stdout:
```
{% include copy.html %}

> When an event contains the following input example:

Reviewer: ```suggestion
When an event contains the following example input:
```

```
{"query" : "open source", "some_other_field" : "abc" }
```

> The input will be split into multiple events based on the field `query`, as the delimiter set as whitespace, as shown in the following example:

Reviewer 2: ```suggestion
The input will be split into multiple events based on the field `query`, as the delimiter set as white space, as shown in the following example:
```

> The input will be split into multiple events based on the field `query`, as the delimiter set as white space, as shown in the following example:

Reviewer: ```suggestion
The input will be split into multiple events based on the `query` field, as the delimiter set as white space, as shown in the following example:
```

> The input will be split into multiple events based on the field `query`, as the delimiter set as white space, as shown in the following example:

Reviewer: "with" the delimiter set to white space (instead of "as")?
